### ISSUE TITLE
Autocorrection for Lint/UnusedMethodArgument should update Sorbet `sig`

### ISSUE THREAD

**Issue by dduugg** [rubocop/rubocop#11021]
--------

## Expected behavior

Per https://github.com/rubocop/rubocop/pull/11020#issuecomment-1258896001 the autocorrect for Lint/UnusedMethodArgument should update the Sorbet type signature, e.g.

```ruby
sig { params(x: Integer).void }
def foo(x); end
```
should autocorrect to:
```ruby
sig { params(_x: Integer).void }
def foo(_x); end
```

(Alternatively, the cop could be marked `SafeAutoCorrect: false`, which was attempted in https://github.com/rubocop/rubocop/pull/11020 )

## Actual behavior

The parameter is renamed under autocorrection, but not the type signature:
```
sig { params(x: Integer).void }
def foo(_x); end
```

This code will fail to typecheck, as well as resulting in a `RuntimeError` on method invocation.

## Steps to reproduce the problem

run `rubocop -a` on this snippet:
```ruby
sig { params(x: Integer).void }
def foo(x); end
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ bundle exec rubocop -V
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.1) [x86_64-darwin21]
```

**Comment by dduugg** [comment]
On further consideration, the autocorrect would still break rbi/rbs files, and I'm not sure rubocop supports the necessary correction across files (and file tops). I think the cop still needs to be marked `SafeAutoCorrect: false`. What do you think, @koic ?

**Comment by corsonknowles** [comment]
I wonder if we could amend this rule to ignore typed files? Perhaps a monkey-patched version in:
* https://github.com/Shopify/rubocop-sorbet
is the most community friendly solution?
